python/hail/dataset.py

```diff
@@ -2422,35 +2450,184 @@ def persist(self, storage_level="MEMORY_AND_DISK"):
             pargs.append(storage_level)
         return self.hc.run_command(self, pargs)
 
-    def print_schema(self, output=None, attributes=False, va=False, sa=False, print_global=False):
-        """Shows the schema for global, sample and variant annotations.
+    def global_schema(self):
+        return Type._from_java(self.jvds.globalSignature())
 
-        :param output: Output file.
-        :type output: str or None
+    def sample_schema(self):
+        return Type._from_java(self.jvds.saSignature())
 
-        :param bool attributes: If True, print attributes.
+    def variant_schema(self):
+        return Type._from_java(self.jvds.vaSignature())
 
-        :param bool va: If True, print variant annotations schema.
+    def query_samples_typed(self, exprs):
```

Comment:
Ah, I see.  I'd put the main documentation on query_samples and just a short description on _typed versions about the difference.